### PR TITLE
Bug: Order of Spool files reverses when the job is Expanded and Collapsed

### PR DESCRIPTION
**Pull request by SanthoshiBoyina**
## Proposed changes
To make spools files order doesn't reverse when the job is expanded and collapsed.

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/job/ZoweJobNode.ts [hunk 1]
```diff
@@ -237,10 +237,14 @@ export class ZoweJobNode extends ZoweTreeNode implements IZoweJobTreeNode {
...
         const sortMethod = contextually.isSession(this) ? this.sort : { method: JobSortOpts.Id, direction: SortDirection.Ascending };
         // Remove any children that are no longer present in the built record
-        this.children = this.children
-            .concat(newChildren)
-            .filter((ch) => Object.values(elementChildren).find((recordCh) => recordCh.label === ch.label) != null)
-            .sort(ZoweJobNode.sortJobs(sortMethod));
+        this.children = contextually.isSession(this)
+            ? this.children
+                  .concat(newChildren)
+                  .filter((ch) => Object.values(elementChildren).find((recordCh) => recordCh.label === ch.label) != null)
```
Comment (traeok) [inline packages/zowe-explorer/src/job/ZoweJobNode.ts:243]: I see that lines 242-243 and 246-247 have the same logic. For readability, could this be reduced by checking if the node has a session before sorting the children array? For example:

```ts
this.children = this.children
    .concat(newChildren)
    .filter((ch) => Object.values(elementChildren).find((recordCh) => recordCh.label === ch.label) != null);
if (contextually.isSession(this)) {
    this.children.sort(ZoweJobNode.sortJobs(sortMethod));
}
```

**Review by traeok — COMMENTED** [review]
Tested and looks good, thanks @SanthoshiBoyina - had a quick question about setting `this.children`, just wanted to get your thoughts before approving 🙂

File: packages/zowe-explorer/src/job/ZoweJobNode.ts (hunk 1 above)
Comment (SanthoshiBoyina) [reply packages/zowe-explorer/src/job/ZoweJobNode.ts:243]: Earlier, I used the above-mentioned approach to escape code redundancy but then the SonarCloud Code Analysis - Maintainability check failed. So I have tried out this way @traeok .

Comment (traeok) [reply packages/zowe-explorer/src/job/ZoweJobNode.ts:243]: That's fair, I didn't realize that SonarCloud had flagged that section of code. Thanks for clarifying!

**Review by traeok — APPROVED** [review]

**Review by adam-wolfe — APPROVED** [review]
LGTM, thanks @SanthoshiBoyina!

**Pull request merged**